### PR TITLE
30 unit tests for candidateadminapi

### LINKED ISSUE #30: Unit tests for CandidateAdminApi

### PR DESCRIPTION
**Pull request by sadatmalik**
- Unit tests for CandidateAdminApi
- Remove unnecessary 'this' references in CandidateAdminApi

### CODE REVIEW

File: server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java [hunk 1]
```diff
@@ -109,42 +109,42 @@ public Map<String, Object> search(@RequestBody SearchCandidateRequest request) {
 
     @PostMapping("findbyemail")
     public Map<String, Object> findByCandidateEmail(@RequestBody CandidateEmailSearchRequest request) {
-        Page<Candidate> candidates = this.candidateService.searchCandidates(request);
+        Page<Candidate> candidates = candidateService.searchCandidates(request);
```
Comment (camerojo) [inline server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java:112]: This is code written by an Angular programmer! (One of the team who ported the code to Sprint/Angular in 2019!)

In Angular, you always have to specify "this"

File: server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java [hunk 2]
```diff
@@ -328,7 +328,7 @@ public String generateToken(@PathVariable("cn") String candidateNumber,
                                 @RequestParam(defaultValue = "false") boolean restrictCandidateOccupations,
                                 @RequestParam(defaultValue = "") List<Long> candidateOccupationIds) {
              CvClaims cvClaims = new CvClaims(candidateNumber, restrictCandidateOccupations, candidateOccupationIds);
-             String token = this.candidateTokenProvider.generateCvToken(cvClaims, 365L);
+             String token = candidateTokenProvider.generateCvToken(cvClaims, 365L);
```
Comment (camerojo) [inline server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java:331]: There was a merge conflict I had to resolve here with changes that Tim Hill checked in. All good now.

**Review by camerojo — APPROVED** [review]
Nice! Looking forward to using this framework.

File: server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java (hunk 2 above)
Comment (sadatmalik) [reply server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java:331]: Thanks John! I pulled from staging before pushing for review but Tim's changes went in yesterday after I had done that - thanks for resolving.

File: server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java (hunk 1 above)
Comment (sadatmalik) [reply server/src/main/java/org/tbbtalent/server/api/admin/CandidateAdminApi.java:112]: Yep, thought so. As it was being used in some places and not in others I thought it best to take the opportunity to make it consistent throughout.

**Pull request merged**